### PR TITLE
openbao 2.0.0-beta20240618 (new formula)

### PR DESCRIPTION
**Pull request by Evengard**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
This adds a new formulae for the openbao package, which should effectively replace the existing vault package in due time

The formulae mimicks pretty closely the vault one

It is a prerelease version because for now openbao still doesn't have a GA version yet.

Still the `--online` flag of the audit command complains about that (and nothing else)

### CODE REVIEW

**Comment by chenrui333** [comment]
@Evengard add this into audit_exceptions/github_prerelease_allowlist.json for now.

File: Formula/o/openbao.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "yarn" => :build
+
+  uses_from_macos "curl" => :test
+
+  def install
+    ENV.prepend_path "PATH", Formula["node"].opt_libexec/"bin" # for npm
+    system "make", "bootstrap", "static-dist", "dev-ui"
```
Comment (SMillerDev) [inline Formula/o/openbao.rb:18]: Does this compile with static libraries? Because it shouldn't.

Comment (Evengard) [reply Formula/o/openbao.rb:18]: Not sure what do you mean. This is how the vault package was made, and seeing as openbao is basically the FOSS fork of vault I used the same.

Do you think it should compile with dynamic libs?

**Comment by Evengard** [comment]
@chenrui333 Added the exception to audit_exceptions/github_prerelease_allowlist.json, now audit passes. Seems like there is no need to add it to audit_exceptions/unstable_allowlist.json?

**Comment by chenrui333** [comment]
> @chenrui333 Added the exception to audit_exceptions/github_prerelease_allowlist.json, now audit passes. Seems like there is no need to add it to audit_exceptions/unstable_allowlist.json?

yeah, looks like no need to have `audit_exceptions/unstable_allowlist.json`

File: Formula/o/openbao.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/o/openbao.rb:18]: yeah, it is the same thing in the vault

**Review by chenrui333 — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]

**Pull request merged**